integrations/java/openhouse-java-itest/src/test/java/com/linkedin/openhouse/javaclient/SmokeTest.java

```diff
@@ -111,6 +118,21 @@ public void testCatalogGetTableWorks() {
         () -> openHouseCatalog.loadTable(TableIdentifier.of("db", "table")));
   }
 
+  @Test
+  public void testCatalogRefreshTableWorks() {
+    mockTableService.enqueue(
+        new MockResponse().setResponseCode(404).addHeader("Content-Type", "application/json"));
+    OpenHouseCatalog openHouseCatalog = new OpenHouseCatalog();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(CatalogProperties.URI, url);
+    properties.put("auth-token", "Test");
+    openHouseCatalog.initialize("openhouse", properties);
+    openHouseCatalog.setToken("Test2");
+    Assertions.assertThrows(
```

Comment:
can we also validate apiClient.getDefaultHeader has the new token after setToken? I'm assuming youll need `VisibilityForTesting`

Reply:
ApiClient does not expose the defaultHeaders which can be read to validate that tokens are set. 
Although I could verify with debugger that the token is updated in ApiClient, I could not do a get on it to assert.

Overall this test validates that setToken does perform updates on ApiClient but lacks assertions.